### ISSUE TITLE
Contrast range for alpha transparent colors

### ISSUE THREAD

**Issue by LeaVerou** [w3c/wcag21#571]
A few years ago I did some research on how to calculate the range of contrast ratios a semi-transparent color can produce and posted it in the WAI IG mailing list:
https://lists.w3.org/Archives/Public/w3c-wai-ig/2012OctDec/0011.html

I also implemented it in my contrast ratio tool: http://leaverou.github.io/contrast-ratio/#white-on-hsla%28200%2C0%25%2C0%25%2C.5%29

Since a new version of WCAG is being worked on, I was wondering if this information could be added. WCAG 2.0 only discusses contrast for opaque colors, but in today’s CSS semi-transparent colors are super common.

**Comment by alastc** [comment]
Hi @LeaVerou,

Thanks for the reminder, it's an interesting problem area.

In terms of what WCAG requires in the normative text, it is whatever the user is seeing, so I don't think that needs to change (unless I'm missing something).

The [understanding doc for text-contrast](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html) seems like a good place for an update, I think the actions would be:

- Add a short section under "Notes on formula" and above "Specific benefits" which notes that transparency has to be taken into account (drawing from your [email](https://lists.w3.org/Archives/Public/w3c-wai-ig/2012OctDec/0011.html)).
- Note which tools take account of transparency (e.g. the CCA does because it is an eye-dropper approach, but some reading the CSS will not).
- Add the [contrast ratio](http://leaverou.github.io/contrast-ratio/#white-on-hsla%28200%2C0%25%2C0%25%2C.5%29) tool.

Does that sound reasonable?

**Comment by LeaVerou** [comment]
Sure!

**Comment by joshueoconnor** [comment]
Ok, this looks like you are happy with @alastc suggestion here, if we can bring this suggestion to the group and close this comment. Is that ok @LeaVerou ?

**Comment by LeaVerou** [comment]
Yup!

**Comment by joshueoconnor** [comment]
Thanks @LeaVerou - @alastc are you ok to bring this as a proposal to the group?

**Comment by alastc** [comment]
@joshueoconnor Sure, it's an edit to a 2.0 understanding doc, I assume that can be done during the 2.1 work though.
